cloud-controller-manager/do/loadbalancers.go

```diff
@@ -275,27 +306,91 @@ func (l *loadBalancers) EnsureLoadBalancer(ctx context.Context, clusterName stri
 	}, nil
 }
 
-// UpdateLoadBalancer updates the load balancer for service to balance across
-// the droplets in nodes.
-//
-// UpdateLoadBalancer will not modify service or nodes.
-func (l *loadBalancers) UpdateLoadBalancer(ctx context.Context, clusterName string, service *v1.Service, nodes []*v1.Node) error {
+func getCertificateIDFromLB(lb *godo.LoadBalancer) string {
+	var id string
+	for _, rule := range lb.ForwardingRules {
+		if rule.CertificateID != "" {
+			id = rule.CertificateID
+			// CCM does not currently support multiple certificates on the
+			// loadbalancers it manages so we only need to grab the first one we find
+			break
+		}
+	}
+	return id
+}
+
+func (l *loadBalancers) checkAndUpdateLBAndServiceCerts(ctx context.Context, service *v1.Service, lbCertID, serviceCertID string) error {
```

Comment:
Method name needs to be updated. Suggestion: `recordLetsEncryptLBCert`

(I've dropped the `check` prefix deliberately, don't feel it's necessary to encode the check-ness nature of code into names as we do those for short-circuiting all the times.)

Reply:
Can we add a GoDoc describing what the function does and in particular why we need it (i.e., explain the case of LE certs getting updated on LBs and CCM needing to sync with that)?